app.py

```diff
@@ -14,6 +14,70 @@ def get_pools():
     return jsonify({"pools": Pool.get_table()})
 
 
+@app.route("/pool", methods=["GET"])
+def get_pool():
+    if "id" not in request.args:
+        return "Pool ID not provided in request", 400
+    id = request.args.get('id')
+    try:
+        pool = Pool.get_pool(id)
+        return jsonify({"pool": Pool.json(pool)})
+    except AttributeError:
+        return "Pool of ID {} doesn't exist".format(id), 404
+
+
+@app.route("/add_pool", methods=["POST"])
+def add_pool():
+    if not request.json:
+        return "Pool data not provided", 400
+    try:
+        pool_id = request.json['ID']
+        pool = Pool.add_pool(pool_id,
+                             request.json['Name'],
+                             request.json.get('MaximumCount', 0),
+                             request.json.get('Description', ''),
+                             request.json.get('Enabled', False))
+        installed_software = request.json.get('InstalledSoftware', [])
+        for name, version in installed_software:
+            sw = Software.add_software(name)
+            pool.add_software(sw, version)
+        return Pool.get_pool(pool_id).ID, 200
+    except KeyError as e:
+        return "Value of {} missing in given JSON".format(e), 400
+
+
+@app.route("/edit_pool", methods=["POST"])
+def edit_pool():
+    if "id" not in request.args:
+        return "Pool ID not provided in request", 400
+    if not request.json:
+        return "Pool data not provided", 400
+
+    id = request.args.get('id')
+    try:
+        if Pool.edit_software(id, request.json.get('InstalledSoftware', [])) \
+               and Pool.edit_pool(id,
+                           request.json['ID'],
+                           request.json['Name'],
+                           request.json.get('MaximumCount', ''),
+                           request.json.get('Description', ''),
+                           request.json.get('Enabled', False)):
+            return "Pool successfully edited", 200
+        return "Problem with editing pool", 422
+    except AttributeError:
+        return "Pool of ID {} doesn't exist!".format(id), 404
+
+
+@app.route("/rm_pool", methods=["GET"])
+def rm_pool():
+    if "id" not in request.args:
+        return "Pool ID not provided in request", 400
+    id = request.args.get('id')
+    if not Pool.rm_pool(id):
```

Comment:
Lepiej tu zrobić "try" i złapać błąd niż robić if not - jak się usuwanie nie uda z innego powodu niż to, że nie ma puli o takim id, to user nie mamy o tym pojęcia